lib/data/blocs/bloc.dart

```diff
@@ -1,49 +1,48 @@
 import 'package:moor/moor.dart';
-
-import 'package:nfsee/models.dart';
 import 'package:nfsee/data/database/database.dart';
+import 'package:nfsee/models.dart';
 
 class NFSeeAppBloc {
   final Database db;
 
-  Stream<List<DumpedRecord>> _dumpedRecords;
-  Stream<List<SavedScript>> _savedScripts;
+  Stream<List<DumpedRecord>>? _dumpedRecords;
+  Stream<List<SavedScript>>? _savedScripts;
 
-  Stream<List<DumpedRecord>> get dumpedRecords => _dumpedRecords;
+  Stream<List<DumpedRecord>>? get dumpedRecords => _dumpedRecords;
 
-  Stream<List<SavedScript>> get savedScripts => _savedScripts;
+  Stream<List<SavedScript>>? get savedScripts => _savedScripts;
 
   NFSeeAppBloc() : db = constructDb() {
     _dumpedRecords = db.watchDumpedRecords();
     _savedScripts = db.watchSavedScripts();
   }
 
   Future<int> addDumpedRecord(String data,
-      [DateTime time, String config]) async {
+      [DateTime? time, String? config]) async {
     return await db.addDumpedRecord(DumpedRecordsCompanion().copyWith(
         data: Value(data),
         time: Value(time ?? DateTime.now()),
         config: Value(config ?? DEFAULT_CONFIG)));
   }
 
-  void updateDumpedRecordConfig(int id, String config) {
+  void updateDumpedRecordConfig(int? id, String config) {
```

Comment:
Why bother using `int?` if you are writing `id!` immediately?

Reply:
Actually, there is no relation between `int?` and `id!`. 
`int ?` indicates that parameter `id` may be empty in function `void updateDumpedRecordConfig( )`.
`id!` indicates that parameter `id` may be empty and needed to be checked in function `db.writeDumpedRecord( )`.
Finally I reset the function `db.writeDumpedRecord( ) `, and conversation is resolved.